### ISSUE TITLE
Additional (secondary/tertiary) Asterisk node config in multinode environment

### ISSUE THREAD

**Issue by szenzero** [chan-sccp/sccp_manager#32]
Hi,

In a multinode environment (2 nodes at least; primary/secondary, or as Cisco calls it, publisher and subcscriber(s)), phones have the ability to register to the primary node and open a TCP/SCCP connection toward the secondary (and tertiary) nodes (3 nodes max. in a call manager group according to Cisco). 
When the primary node fails (TCP or SCCP keepalives are lost), phone reverts to next one in the CM group.
More about it here:
https://github.com/chan-sccp/chan-sccp/wiki/Keepalive-and-Failover-Architecture and 
https://github.com/chan-sccp/chan-sccp/wiki/registration,-keepalive-and-srst

Additional callmanagers should be added to the phone config xml (two node example). processNodeName, name, description should be changed accordingly:
```
  <devicePool>
    <callManagerGroup>
      <members>
        <member priority="0">
          <callManager>
            <ports>
              <ethernetPhonePort>2000</ethernetPhonePort>
            </ports>
            <processNodeName>192.168.1.228</processNodeName>
            <name>Asterisk</name>
            <description>Asterisk</description>
          </callManager>
        </member>
        <member priority="1">
          <callManager>
            <ports>
              <ethernetPhonePort>2000</ethernetPhonePort>
            </ports>
            <processNodeName>192.168.1.229</processNodeName>
            <name>Asterisk2</name>
            <description>Asterisk2</description>
          </callManager>
        </member>
      </members>
    </callManagerGroup>
  </devicePool>
```

Screenshot from the Cisco IP Communicator softphone (Settings -> Device configuration) and two node asterisk setup:
![Capture](https://user-images.githubusercontent.com/50993009/134303648-2e39040d-634b-46d6-8e5b-f85a61ac86eb.PNG)

**Comment by szenzero** [comment]
Found a way to configure multiple CM nodes in phone xml files (sort of)... The setting we need here is called "Device Connect Address / Name" found in Server config -> Site Default Settings -> SCCP Networks

![Snimka zaslona 2021-09-28 180553](https://user-images.githubusercontent.com/50993009/135125062-c030b507-619c-44d2-8d19-be7d3dd1a52a.jpg)

So, to add two CM nodes (primary and secondary), you have to insert the first node IP address, then add (+ sign) the second one. But, looks like something seems to be broken while inserting IPs to the DB. Clicking the save button (and after the page auto reloads), shows a strange entry in the "Device Connect Address" field:

![Snimka zaslona 2021-09-28 182123](https://user-images.githubusercontent.com/50993009/135127067-1086f06a-bd14-4c54-9d7f-41e2f4fe2634.jpg)

If you check the DB, you'll see sccpconfig.ccm_address value is populated with IPs inserted above but without any delimiter (should be semicolon by looking at the code)

```
select * from sccpsettings where keyword='ccm_address'\G;
*************************** 1. row ***************************
      keyword: ccm_address
          seq: 98
         type: 2
         data: 5.5.5.5:20006.6.6.6:2000
systemdefault: 
```

Could be quickfixed by updating the sccpconfig.ccm_address value as follows:
`update sccpsettings set data = '5.5.5.5:2000;6.6.6.6:2000' where keyword='ccm_address';` 

Config is now correctly parsed:
![Snimka zaslona 2021-09-28 182725](https://user-images.githubusercontent.com/50993009/135127634-f9c1064d-985f-4cb2-8e62-5f820557ca6c.jpg)

And phone XML is somewhat successfully generated. Port number (2000) is added as a part of "processNodeName" param. (which should be IP only)


```
   <callManagerGroup>
      <members>
        <member priority="0">
          <callManager>
            <ports>
              <ethernetPhonePort>2000</ethernetPhonePort>
            </ports>
            <processNodeName>5.5.5.5:2000</processNodeName>
            <name>Asterisk</name>
            <description>Asterisk</description>
          </callManager>
        </member>
        <member priority="1">
          <callManager>
            <ports>
              <ethernetPhonePort>2000</ethernetPhonePort>
            </ports>
            <processNodeName>6.6.6.6:2000</processNodeName>
            <name>Asterisk</name>
            <description>Asterisk</description>
          </callManager>
        </member>
      </members>
    </callManagerGroup>
```

Manually changing the sccpconfig.ccm_address should do the trick:
`update sccpsettings set data = '5.5.5.5;6.6.6.6' where keyword='ccm_address';`

And the generated XML:
```
<callManagerGroup>
      <members>
        <member priority="0">
          <callManager>
            <ports>
              <ethernetPhonePort>2000</ethernetPhonePort>
            </ports>
            <processNodeName>5.5.5.5</processNodeName>
            <name>Asterisk</name>
            <description>Asterisk</description>
          </callManager>
        </member>
        <member priority="1">
          <callManager>
            <ports>
              <ethernetPhonePort>2000</ethernetPhonePort>
            </ports>
            <processNodeName>6.6.6.6</processNodeName>
            <name>Asterisk</name>
            <description>Asterisk</description>
          </callManager>
        </member>
      </members>
    </callManagerGroup>
```

**Comment by szenzero** [comment]
I guess this could be considered as a bug report in the v14.2 branch (as the enhancement/feature is already there: multiple "Device connect address" entries in server config)...

I've just installed the develop branch module and adding multiple device connect addresses as well as "allow networks", "deny networks", "local networks" addresses work as expected.

**Comment by steve-lad** [comment]
Hi, so can you confirm that as of 14.3 (->14.4 via module admin) - that the required functionality is already there?

If so, are you suggesting improved naming/ help for Device Connect Address?

BTW 14.2 is dead - it was an initial version from PhantomVL, but has long since been replaced and the current stable branch is v14.4 - I will be updating the WiKi shortly

**Comment by szenzero** [comment]
Yes, the functionality is there (Device Connect Address), seems like a bug in GUI/backend. If you try to add the secondary address (via the plus sign)
![image](https://user-images.githubusercontent.com/50993009/150518792-e2279702-edf2-41a4-910a-1da39680b6df.png)
after saving, you'll get something like this:
![image](https://user-images.githubusercontent.com/50993009/150518902-697c9a52-38c6-40d4-bcfc-5fe7c99853f7.png)
The value in DB will also be incorrect as well as the generated phone config files.

Tried with version 14.4.0.5; upgraded via CLI.

**Comment by steve-lad** [comment]
OK thanks for clarifying - will have a look

**Comment by steve-lad** [comment]
Thanks for reporting this. It is fixed in 14.5.0.1
